src/libstd/sys/unix/ext/net.rs

```diff
@@ -213,7 +213,9 @@ impl SocketAddr {
 
     fn address<'a>(&'a self) -> AddressKind<'a> {
         let len = self.len as usize - sun_path_offset();
-        let path = unsafe { mem::transmute::<&[libc::c_char], &[u8]>(&self.addr.sun_path) };
+        let path: &[libc::c_char] = unsafe {
+            &*(&self.addr.sun_path as *const _ as *const _)
```

Comment:
This one is actually wrong — the original is tricky. `self.addr.sun_path` is an array and the transmute with explicit type is using the coercion from array to slice. This doesn't happen when we cast the reference to array in the new code.

I think the elided types were close to obscuring this bug.

Reinterpreting `&[u8] <-> &[c_char]` seems like it should be a function.

Reply:
And the old target type was `&[u8]`, in this code it is instead `&[c_char]`. They should switch places. Convert from c_char to u8.

Reply:
I see what you mean. Thanks a lot @bluss, you're awesome 👍🏻. Thanks for everything you do for this project. I'm make the appropriate changes tomorrow.